CMakeLists.txt

```diff
@@ -323,6 +324,13 @@ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 add_compile_definitions(${HOST_OS} PACKAGE_NAME="ats" PACKAGE_VERSION="${TS_VERSION_STRING}")
 add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-invalid-offsetof>)
 
+# Enable fuzzing
+if(ENABLE_FUZZING)
+    if(NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
Why `MATCHES` instead of `STREQUAL`?

Reply:
> Why MATCHES instead of STREQUAL?

Ah My bad, I can change this to `STREQUAL`.

Reply:
`MATCHES` is used for regular expression comparisons. If what you really want is a string comparison, then `MATCHES` is doing more than you intended and is sloppy from a performance and readability perspective. It could also introduce subtle unexpected behavior: any compiler that happens to have "Clang" in the name will match. It's also possible that you want that behavior - maybe you want Apple Clang to count. Whichever way you intend to do it, you want to understand why you did it that way and which alternatives you decided against and why.

Reply:
>  It could also introduce subtle unexpected behavior: any compiler that happens to have "Clang" in the name will match. 

I understand, But this is  what I want.

When [AFL](https://github.com/AFLplusplus/AFLplusplus) is used for fuzzing, the compilers are:

```bash
CC=/src/aflplusplus/afl-clang-fast
CXX=/src/aflplusplus/afl-clang-fast++
```

Doing it the other way around will break the AFL fuzzer build.


#### My comment:

> if(NOT CMAKE_CXX_COMPILER_ID MATCHES "Clang")

I don't wan't `AFL` and `honggfuzz`  build to break.

NOTE:  AFL and Honggfuzz dose support GCC, but GCC instrumentation is kind of crap.

Reply:
Thanks for thinking through it with me. All the CMake changes look good to me.